### PR TITLE
Dialtone V6 Components Radio & Checkbox Implementations

### LINKED ISSUE #215: [CSS] Component: Checkboxes
**Acceptance Criteria**

- Update the checkbox component Dialtone classes

**Resources**

Design: https://www.figma.com/file/zP2uyVxAg4notGNZwW64mW/Components-(Dialtone-6.0)?node-id=389%3A0

### LINKED ISSUE #220: [CSS] Component: Radios
**Acceptance Criteria**

- Update the radio component Dialtone classes

**Resources**

Design: https://www.figma.com/file/zP2uyVxAg4notGNZwW64mW/Components-(Dialtone-6.0)?node-id=3845%3A7529

### PR DESCRIPTION
**Pull request by ryandialpad**
# Dialtone V6 Components Radio & Checkbox Implementations

## :hammer_and_wrench: Type Of Change

- [ ] Fix
- [x] Feature
- [x] Refactoring
- [ ] Documentation

## :book: Description

Updated radio and checkbox styles
Light updates to the documentation

## :bulb: Context

Closes #215
Closes #220

## Pull Request Checklist

 - [x] Ask the contributors if a similar effort is already in process or has been solved.
 - [x] Review the [contribution guidelines](https://github.com/dialpad/dialtone/blob/staging/.github/CONTRIBUTING.md).
 - [x] Ensure all `gulp` scripts successfully compile.
 - [x] Update, remove, or extend all affected documentation.
 - [x] Provide a description what is being changed, extended, or removed.
 - [x] Link to any issue(s) this request is resolving.
 - [x] Ensure no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
 - [x] Request a review from Drew Chandler, Joshua Hynes, or Ted Goas.

### Checkboxes

Screen Shot 2021-03-18 at 2 08 22 PM
Screen Shot 2021-03-18 at 2 08 30 PM
Screen Shot 2021-03-19 at 2 31 45 PM
Screen Shot 2021-03-18 at 2 08 47 PM

### Radios

Screen Shot 2021-03-18 at 2 09 02 PM
Screen Shot 2021-03-18 at 2 09 10 PM
Screen Shot 2021-03-19 at 2 31 57 PM

### CODE REVIEW

File: lib/less/components/links.less [hunk 1]
```diff
@@ -52,14 +52,15 @@
 //  $$  WARNING
```
Comment (ryandialpad) [inline lib/less/components/links.less]: Changes from a merged commit, will disappear once links component is merged into version6 and then merged into the component branch

File: lib/less/variables/colors.less [hunk 2]
```diff
@@ -191,15 +191,19 @@
     //  --  STATES
```
Comment (ryandialpad) [inline lib/less/variables/colors.less]: Changes from a merged commit, will disappear once links component is merged into version6 and then merged into the component branch

File: lib/less/components/radio-checkbox.less [hunk 3]
```diff
@@ -25,9 +25,11 @@
 .d-radio {
     //  Component specific CSS Vars
     //  ------------------------------------------------------------------------
-    --check-radio--bc:      var(--black-600);
-    --check-radio--bgc:     var(--white);
-    --check-radio--bs:      none;
+    --check-radio--color:         var(--primary-color);
+    --check-radio--color-focus:   hsla(var(--primary-color-hsl) ~" / " 15%);
```
Comment (ryandialpad) [inline lib/less/components/radio-checkbox.less:29]: Anyone object to these variable names? They are used to reduce the amount of duplicated code

File: lib/less/components/radio-checkbox.less [hunk 4]
```diff
@@ -113,7 +122,7 @@
 .d-radio__copy {
     display: inline-flex;
     flex-direction: column;
-    flex: 1 auto;
```
Comment (ryandialpad) [inline lib/less/components/radio-checkbox.less:116]: Resolved issues outlined in #196

File: lib/less/components/radio-checkbox.less [hunk 5]
```diff
@@ -124,9 +133,12 @@
     flex-direction: column;
     flex: 1 auto;
 
+    color: var(--black-800);
     font-size: var(--fs14);
     line-height: calc(20/14);
     font-weight: var(--fw-normal);
+
+    cursor: pointer;
```
Comment (ryandialpad) [inline lib/less/components/radio-checkbox.less:141]: Resolved issues outlined in #196

File: lib/less/components/radio-checkbox.less [hunk 6]
```diff
@@ -215,100 +234,68 @@
...
-    }
-    &:checked {
-        border-color: var(--success-color);
-        background-color: var(--success-color);
-        box-shadow: inset 0 0 0 var(--su2) var(--white);
-    }
+.d-radio--success {
+    --check-radio--color:         var(--success-color);
+    --check-radio--color-focus:   var(--success-color-focus);
+    --check-radio--bc:            var(--check-radio--color);
```
Comment (ryandialpad) [inline lib/less/components/radio-checkbox.less:302]: Validation states now only modify the colors which are applied the the radio / checkbox

File: lib/less/components/radio-checkbox.less [hunk 7]
```diff
@@ -215,100 +234,68 @@
...
 .d-checkbox--success {
-    --check-radio--bc:      var(--success-color);
-
-    &:focus,
-    &:checked:focus {
-        --check-radio--bs:  0 0 0 var(--su4) hsla(var(--success-color-hsl) ~" / " 25%);
-    }
+    --check-radio--color:         var(--success-color);
+    --check-radio--color-focus:   var(--success-color-focus);
 }
```
Comment (ryandialpad) [inline lib/less/components/radio-checkbox.less]: Validation states now only modify the colors which are applied the the radio / checkbox

File: lib/less/components/radio-checkbox.less [hunk 8]
```diff
@@ -52,14 +54,15 @@
         font-size: inherit;
         cursor: pointer;
 
-        &:focus {
-            --check-radio--bc: var(--primary-color);
-            --check-radio--bs: 0 0 0 var(--su4) hsla(var(--primary-color-hsl) ~" / " 15%);
+        &:focus,
+        &:checked:focus {
+            --check-radio--bc: var(--check-radio--color);
+            --check-radio--bs: 0 0 0 var(--su1) var(--white), 0 0 0 var(--su4) var(--check-radio--color-focus);
```
Comment (ryandialpad) [inline lib/less/components/radio-checkbox.less:60]: Updates to match new design

File: lib/less/components/radio-checkbox.less [hunk 9]
```diff
@@ -215,100 +234,68 @@
...
+            --check-radio--bc:      var(--black-400);
+            --check-radio--bgc:     var(--black-200);
+        }
+
         &:focus,
         &:checked:focus {
-            --check-radio--bgc:     var(--primary-color);
-            --check-radio--bs:      inset 0 0 0 var(--su2) var(--white), 0 0 0 var(--su4) hsla(var(--primary-color-hsl) ~" / " 15%);
+            --check-radio--bgc:     var(--check-radio--color);
+            --check-radio--bs:      0 0 0 var(--su1) var(--white), inset 0 0 0 var(--su2) var(--white), 0 0 0 var(--su4) var(--check-radio--color-focus);
```
Comment (ryandialpad) [inline lib/less/components/radio-checkbox.less:270]: Updates to match new design

File: docs/components/checkboxes.html [hunk 10]
```diff
@@ -27,15 +27,15 @@ <h2 class="dialtone-title__section">Base Style</h2>
                             <label class="d-checkbox__label" for="Dialtone-CheckExample1">Checkbox label</label>
                         </div>
                     </div>
-                    <div class="d-checkbox-group--disabled">
+                    <div class="d-checkbox-group d-checkbox-group--disabled">
```
Comment (ryandialpad) [inline docs/components/checkboxes.html:30]: Correction: Classes need to be chained

File: docs/components/radios.html [hunk 11]
```diff
@@ -27,15 +27,15 @@ <h2 class="dialtone-title__section">Base Style</h2>
                             <label class="d-radio__label" for="Dialtone-RadioExample1">Radio label</label>
                         </div>
                     </div>
-                    <div class="d-radio-group--disabled">
+                    <div class="d-radio-group d-radio-group--disabled">
```
Comment (ryandialpad) [inline docs/components/radios.html:30]: Correction: Classes need to be chained

File: lib/less/components/radio-checkbox.less [hunk 12]
```diff
@@ -191,22 +209,23 @@
...
             //  Disabled
             &[disabled] {
-                --check-radio--bc:      var(--black-300);
-                --check-radio--bgc:     var(--black-100);
-            }
-
-            &:focus {
-                --check-radio--bs:      0 0 0 var(--su4) hsla(var(--primary-color-hsl) ~" / " 25%);
+                --check-radio--bgc:     var(--black-200);
+                background-image: url("data:image/svg+xml,%3Csvg width='12' height='12' viewBox='0 0 24 24' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z' fill='hsl(240, 10%, 51%)'/%3E%3C/svg%3E");
```
Comment (ryandialpad) [inline lib/less/components/radio-checkbox.less:227]: Updated SVG to match updated design

**Comment by dialpad-drew** [comment]
@ryandialpad   I know you're breaking it up so that validation messages are a separate component, but would it possible to include it in this PR so we review these inputs' validation states holistically?

One thing I did notice, though: it appears as though Checkbox's default border color is blue, when it should be a black. Both radio's and checkbox's input border color should be `--black-500`.

**Comment by ryandialpad** [comment]
@dialpad-drew Updated border color from black 600 to black 500. Fixed checkbox border to not display blue swatch by default

**Review by dialpad-drew — COMMENTED** [review]
These looks good. Only thing I'm noticing are in the videos below -- I'm not sure what this is related to yet, so I'm curious if you see it on your side as well.

https://user-images.githubusercontent.com/77751297/112028989-62de6380-8afe-11eb-8b78-5e9e7f96f6a7.mov

https://user-images.githubusercontent.com/77751297/112028982-6114a000-8afe-11eb-9669-a9d058dd6668.mov

**Comment by ryandialpad** [comment]
@dialpad-drew Good catch! Will investigate the issue and report back

**Comment by ryandialpad** [comment]
@dialpad-drew I was only able to replicate the scenario with the top most example. It might be related to how the example is built, will keep you posted

**Comment by ryandialpad** [comment]
@dialpad-drew Found the issue, the labels are tied to the incorrect inputs (the top example has both labels pointing at the first input). Fix to come shortly

**Comment by ryandialpad** [comment]
@dialpad-drew Fixes to the documentation have been pushed, feel free to have a look at your leisure :D

**Review by dialpad-drew — APPROVED** [review]

**Pull request merged**